### DISCUSSION TITLE
Flash encryption and boot secure V2 support?

### DISCUSSION THREAD

**Discussion by dzungpv** [revk/xESP32-Faikout#116] [General]
I test your lib to test it can use with my IDF project. But seem secure boot v2 and flash encryption not working fine. I post it here because not a bug.
This is the log in the dev mode for Faikin app:

```
rst:0x3 (SW_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:2, clock div:2
secure boot v2 enabled
secure boot verification succeeded
load:0x3fff00b8 len:0x2d6c
load:0x40078000 len:0x5ea8
load:0x40080400 len:0x4
0x40080400: _init at ??:?

load:0x40080404 len:0xdb0
entry 0x400805e4
```
It may relate to NVS after enable flash encryption

**Comment by revk** [comment]
S3 had to use a different secret key, if that helps.

**Reply by dzungpv** [reply]
I use ESP-WROOM 32, V3.1 the latest model, other sample code work fine with secure boot v2 and flash encryption but not yours.

**Reply by revk** [reply]
If you have any clue why, do let me know. Seems odd.

**Reply by dzungpv** [reply]
I use same key for boot loader and flash encryption, RSA 3072. Success with other project.

**Reply by dzungpv** [reply]
Yes, I will try to connect to JTAG to check it

**Reply by dzungpv** [reply]
I change CONFIG_REVK_PARTITION_CHECK = no and it boot, so the code relate to it may cause problem, when flash encrypt enable and secure boot CONFIG_PARTITION_TABLE_OFFSET must change and it must be CONFIG_PARTITION_TABLE_OFFSET + 0X1000 for any partition other than boot loader: https://docs.espressif.com/projects/esp-idf/en/release-v4.2/esp32/security/secure-boot-bootloader-size.html I will test and may send a PR to fix the issue

**Reply by manhst** [reply]
@dzungpv chào bác, bác cho em xin contact để em hỏi chút được không ạ :(

**Reply by dzungpv** [reply]
@revk I fix the CONFIG_REVK_PARTITION_CHECK and it can boot with config page,  but setting not save , it show: save 5 setting when set wifi and mqtt but reboot empty, so it may relate to the api you use, not support flash encryption.

**Comment by dzungpv** [comment]
> @dzungpv chào bác, bác cho em xin contact để em hỏi chút được không ạ :(

No, English only, create issue or discussion here if I know I can answer, do not spam unrelated question.
